src/hera/workflows/script.py

```diff
@@ -606,27 +617,81 @@ def _output_annotations_used(source: Callable) -> bool:
 
 FuncIns = ParamSpec("FuncIns")  # For input types of given func to script decorator
 FuncR = TypeVar("FuncR")  # For return type of given func to script decorator
+FuncRCov = TypeVar("FuncRCov", covariant=True)
 
 ScriptIns = ParamSpec("ScriptIns")  # For input attributes of Script class
-StepIns = ParamSpec("StepIns")  # # For input attributes of Step class
-TaskIns = ParamSpec("TaskIns")  # # For input attributes of Task class
 
 
-# Pass actual classes of Script, Step and Task to bind inputs to the ParamSpecs above
+class _ScriptDecoratedFunction(Generic[FuncIns, FuncRCov], Protocol):
```

Comment:
question. I want to make sure I understood alright or not.
Script has lots of arguments than this class is overloading. And doesn't seem to be covered through typevar. are these arguments are selected to resolve type issues? or any other reason for this?

Reply:
Only the parameters to Step and Task are supported here, not the full set of arguments to Script (those are instead used at a higher level in the decorator call chain, specifically they are accepted by `@script` itself).